### PR TITLE
Backlog clearing: 8 months of SC updates! 2023-11 through 2024-06-19

### PR DESCRIPTION
**Pull request by gpshead**
Keep in mind that most of these are "historical" at this point.  I grouped them roughly into "quarters" and tried to coalesce things within those to keep them shorter. In older notes, some of the things mentioned may have been resolved in future sessions.

Notes after these will be posted by our new PSF communications liaison.

### CODE REVIEW

File: updates/2024-04,05,06-steering-council-update.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# April through June 19, 2024 Steering Council update
+
+- Steering Council discussion topics:
+    - **PEPs**
+        - Ongoing [PEP 741, a Configuration C API](https://peps.python.org/pep-0741/) discussions; seeking consensus and clarity and WG input. Ultimately decided to postpone this until 3.14.
+        - Repeated check-ins on [PEP 649, deferred evaluation of annotations using descriptors](https://peps.python.org/pep-0649/) implementation. We're collectively okay if this slips into 3.14.
+        - Finalized and posted our decision on [PEP 734, subinterpreters in the stdlib](https://discuss.python.org/t/pep-734-multiple-interpreters-in-the-stdlib/41147/24).
+        - Accepted [PEP 667, Consistent views of namespaces](https://peps.python.org/pep-0667/) after much discussion. The alternatives were considered worse in the long run.
+    - Sync up meetings with Łukasz, the Developer-in-Residence:
+        - The untangling of potential license and CLA issues on the infrastructure repositories. Not glamourous work, needs doing, will involve PSF council as appropriate.
```
Comment (hugovk) [inline updates/2024-04,05,06-steering-council-update.md:10]:
```suggestion
        - The untangling of potential license and CLA issues on the infrastructure repositories. Not glamorous work, needs doing, will involve PSF counsel as appropriate.
```

File: updates/2024-04,05,06-steering-council-update.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    - Discussed how it'd be nice if our monthly SC update summaries were actually created, reviewed, and published before PyCon US 2024. *(editors note: LOL... given the backlog this is being posted as part of, volunteer time is precious, thus hiring)*
+    - Discussed new 3.13 REPL platform implementation woes. "At least it is not using readline" also means "people who rely on obscure readline featureitis frown".
+    - Discussions of type annotations in the stdlib. More community discussions needed before any proposal can be made. Annotations in `Tools/` have found actual bugs.
+    - Discussed capacity issues at the upcoming September core dev sprint; space limitations are real and lead to hard choices.
+    - Discussed logistics of the remaining two core dev. mentorship sessions that Edward is holding this quarter. *(editors note: held)*
+    - Discussed further ongoing core dev mentorship needs and the utility of a possible Mentorship WG. One theme has always been lack of volunteer time availability.
+    - Discussed how it'd be good if release managers wrote up retrospectives. That requires extra time committment on their part. We'd all benefit, not just future RMs.
```
Comment (hugovk) [inline updates/2024-04,05,06-steering-council-update.md:33]:
```suggestion
    - Discussed how it'd be good if release managers wrote up retrospectives. That requires extra time commitment on their part. We'd all benefit, not just future RMs.
```

File: updates/2024-01,02,03-steering-council-update.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        - Collected job applications and reviewed resumes.
+        - At least 20 people applied!
+        - Interviewed N candidates, rejected the others. Made a choice and sent a job offer.
+        - *Editors note: Our initial job. req. described this role as "Secretary". We all agreed that role title was not great and settled upon a much more appropriate title of "Communications Liaison" with the individual hired.*
+    - Pablo coordinating with Edward for our core dev. mentorship training sessions.
+        - The first mentoring session happened in March. Observation: We *also* used the first one as experience sharing talk time. Note to future selves: We could set our own mini-sessions up just for that.
+    - Lack of use of our SC Office Hours slot and the need to better promote those.
+    - **PEPs**
+        - Rejected [PEP 726 module `__setattr__` and `__delattr__`](https://peps.python.org/pep-0726/) after reviewing it and the resulting discussions. Drafted and posted the pronouncement.
+        - Discussed Windows free-threaded [PEP-703](https://peps.python.org/pep-0703/) installer design details.
```
Comment (hugovk) [inline updates/2024-01,02,03-steering-council-update.md:16]:
```suggestion
        - Discussed Windows free-threaded [PEP 703](https://peps.python.org/pep-0703/) installer design details.
```

File: updates/2024-01,02,03-steering-council-update.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        - Ongoing [PEP 541 Package Index Name Retention](https://peps.python.org/pep-0541/) work.
+        - The status of [PEP 649 deferred eval of annotations](https://peps.python.org/pep-0649/).
+        - Community misconceptions revealed when when talking at a small conference about our upcoming Python 3.13.
+        - The PyCon 2024 US language summit logistics.
+        - The need to provide feedback to our DiRs.
+    - More thoughts on what we want our PyPI org policy to be.
+    - Chatting with Itamar regarding Meta's September 2024 core dev sprint logistics and PSF finance details.
+    - Release Management automation and contingency plan ideas.
+    - The potential formation of a Packaging Council.
+    - Tier support for WASI or WASM and updating PEP-11 to reflect that.
```
Comment (hugovk) [inline updates/2024-01,02,03-steering-council-update.md:47]:
```suggestion
    - Tier support for WASI or WASM and updating [PEP 11](https://peps.python.org/pep-0011/) to reflect that.
```

File: updates/2024-01,02,03-steering-council-update.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+        - Brandt joined us for office hours to discuss this.
+        - A desire for to explain what maintaining and testing entails for fellow core devs.
+        - Result: [PEP 744 JIT Compilation](https://peps.python.org/pep-0744/).
+        - Discussions of where it could head in the future and the need to do those in public forums.
+        - Experimental exit criteria, debugging, profiling, etc.
+    - Started discussing what we should focus on in our PyCon US 2024 keynote slot.
+    - A potential need to make role boundaries more clear between our different groups such as core devs, developers in residence, and release managers. Responsibility becomes more complex as we grow and mix volunteers with paid staff.
+    - An update from the PSF on the status of sponsorships and future needs.
+    - The desire for a better understanding of PyPI's operating costs (from the PSF) and a desire for a roadmap to facilitate better support and sponsorship.
+    - Deb, PSF executive director, channelling the Python users vibe from many angles around packaging ecosystem not being great on many fronts based on different needs and what could be done by whom.
```
Comment (hugovk) [inline updates/2024-01,02,03-steering-council-update.md:58]:
```suggestion
    - Deb, PSF executive director, channelling the Python users' vibe from many angles around packaging ecosystem not being great on many fronts based on different needs and what could be done by whom.
```

File: updates/2024-04,05,06-steering-council-update.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    - Discussed how it'd be nice if our monthly SC update summaries were actually created, reviewed, and published before PyCon US 2024. *(editors note: LOL... given the backlog this is being posted as part of, volunteer time is precious, thus hiring)*
+    - Discussed new 3.13 REPL platform implementation woes. "At least it is not using readline" also means "people who rely on obscure readline featureitis frown".
+    - Discussions of type annotations in the stdlib. More community discussions needed before any proposal can be made. Annotations in `Tools/` have found actual bugs.
+    - Discussed capacity issues at the upcoming September core dev sprint; space limitations are real and lead to hard choices.
+    - Discussed logistics of the remaining two core dev. mentorship sessions that Edward is holding this quarter. *(editors note: held)*
```
Comment (hugovk) [inline updates/2024-04,05,06-steering-council-update.md:31]:
```suggestion
    - Discussed logistics of the remaining two core dev mentorship sessions that Edward is holding this quarter. *(editors note: held)*
```

File: updates/2024-01,02,03-steering-council-update.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+- Steering Council (new 2024 edition) discussion topics:
+    - Discussed our current meeting time, dev in res sync times, and office hours scheduling.
+    - Hiring a part time person for note taking and what our expectations for this role should be.
+        - Drafted and posted [our official job description](https://discuss.python.org/t/the-steering-council-is-hiring/44686).
+        - Collected job applications and reviewed resumes.
+        - At least 20 people applied!
+        - Interviewed N candidates, rejected the others. Made a choice and sent a job offer.
+        - *Editors note: Our initial job. req. described this role as "Secretary". We all agreed that role title was not great and settled upon a much more appropriate title of "Communications Liaison" with the individual hired.*
+    - Pablo coordinating with Edward for our core dev. mentorship training sessions.
```
Comment (hugovk) [inline updates/2024-01,02,03-steering-council-update.md:11]:
```suggestion
    - Pablo coordinating with Edward for our core dev mentorship training sessions.
```

File: updates/2024-04,05,06-steering-council-update.md [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    - Discussed our recent core dev promotion votes. No vetoes (all approved). We're noticing that our process feels a bit odd, the SC represents the community thus a veto of any form would not be well received. Nor would a vote itself failing. So this means anyone proposing someone effectively needs to do homework and talk via backchannels to a slew of core devs and SC first to ensure that neither negative outcome is likely. Maybe this is working as intended? But we don't state it as such anywhere... But what would be meaningfully better to propose?
+    - Discussed and announced who would be our next Release Manager (RM) and related logistical issues.
+    - Continual brainstorming on what to include in what form for our PyCon US 2024 keynote address. In classic PyCon fashion our collective presentation was congealing up until the day we gave it.
+    - Discussed a PR brought to our attention that we believe should not have been merged as is, regardless of intent, and whether or not we can do anything about it right now.
+    - More discussions community friction regarding Packaging and PyPI, the Packaging WG, and PyPA with the PSF.
```
Comment (hugovk) [inline updates/2024-04,05,06-steering-council-update.md:38]:
```suggestion
    - More discussions on community friction regarding Packaging and PyPI, the Packaging WG, and PyPA with the PSF.
```

File: updates/2024-04,05,06-steering-council-update.md [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+    - Noticing how we have misaligned cadences among many of our collective core dev working styles. It leads to a strain on decision making. Perhaps a discussion topic for the core dev sprint at Meta?
+    - Greg and Barry employer changes; no [PEP-13 conflicts](https://peps.python.org/pep-0013/#conflicts-of-interest).
```
Comment (hugovk) [inline updates/2024-04,05,06-steering-council-update.md:40]:
```suggestion
    - Greg and Barry employer changes; no [PEP 13 conflicts](https://peps.python.org/pep-0013/#conflicts-of-interest).
```

**Review by hugovk — COMMENTED** [review]
Thank you for the updates!

CI failure is about updating the index in the README.

Either run `make check` to update it, or apply:

```diff
diff --git a/README.md b/README.md
index 7bbb285..a73d7dd 100644
--- a/README.md
+++ b/README.md
@@ -77,6 +77,9 @@ for filename in filenames:
 - [2023-08](updates/2023-08-steering-council-update.md)
 - [2023-09](updates/2023-09-steering-council-update.md)
 - [2023-10](updates/2023-10-steering-council-update.md)
+- [2023-11,12](updates/2023-11,12-steering-council-update.md)
+- [2024-01,02,03](updates/2024-01,02,03-steering-council-update.md)
+- [2024-04,05,06](updates/2024-04,05,06-steering-council-update.md)
 <!-- [[[end]]] -->
 
 ## Process
```

File: updates/2024-01,02,03-steering-council-update.md [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+        - Logistics of our ability to implement *required* code reviews on new-feature PRs. We have a long term desire to see this for all PRs, but need to prove to core devs that it can work on a smaller uncontroverisal scale first.
+        - Future Release Manager (RM) role planning.
+        - Buildbots and their maintenance issues.
+        - The upgrade of speed.python.org.
+        - Security releases and challenges with 3.8, 3.9, & 3.10.
+        - Challenges and concerns around PyPI's account recovery process and the impact on sponsors and the community. *(editors note: The PSF has since hired someone to help with this!)*
+        - The importance of transparency and good communication for the health of the community.
+        - Ongoing [PEP 541 Package Index Name Retention](https://peps.python.org/pep-0541/) work.
+        - The status of [PEP 649 deferred eval of annotations](https://peps.python.org/pep-0649/).
+        - Community misconceptions revealed when when talking at a small conference about our upcoming Python 3.13.
```
[the remark this answers is not in the document]
Comment (ncoghlan) [reply updates/2024-01,02,03-steering-council-update.md:40]:
```suggestion
        - Community misconceptions revealed when talking at a small conference about our upcoming Python 3.13.
```

**Comment by gpshead** [comment]
Thanks for all the reviews. _(Amusing typos, misspellings, and inconsistent formatting galore.)_

**Pull request merged**